src/Arcus.Templates.WebApi/Program.cs

```diff
@@ -157,10 +151,10 @@ private static void ConfigureServices(WebApplicationBuilder builder, IConfigurat
 
 #if SharedAccessKeyAuth
                 #warning Please provide a valid request header name and secret name to the shared access filter
-                options.Filters.Add(new SharedAccessKeyAuthenticationFilter(headerName: SharedAccessKeyHeaderName, queryParameterName: null, secretName: "<your-secret-name>"));
+                options.Filters.AddSharedAccessKeyAuthenticationOnHeader(SharedAccessKeyHeaderName, "<your-secret-name>");
```

Comment:
Yeah, we should be careful of hiding too much, though, we may choose to add `Filter` to the extension in that case, as the MVC options spans a lot of stuff and it would maybe become too unclear on what this extension is adding.
But, like you said, it's indeed in another project, so we could add an issue there?
Also: since we're looking into using a different take on the authentication filters by using the new .NET 6 `IAuthenticationHandler`, we may want to discard this as we will have other extensions entirely?

Reply:
Yes, we can create an additional issue on the WebApi repo.

Regarding the IAuthenticationHandler topic, you're referring to this discussion: https://github.com/arcus-azure/arcus.webapi/discussions/287 ?

Reply:
I created a new issue here: https://github.com/arcus-azure/arcus.webapi/issues/329

Reply:
> Yes, we can create an additional issue on the WebApi repo.

Great!
 
> Regarding the IAuthenticationHandler topic, you're referring to this discussion: [arcus-azure/arcus.webapi#287](https://github.com/arcus-azure/arcus.webapi/discussions/287) ?

Yeah, that's the one. Seems like something we should investigate further. Already did some but couldn't figure out yet what authentication result we should return as these new authentication handlers forces us to use claims and authenticated users, and in our case with shared access key authentication and the like, there's no such concepts (I think)?

Reply:
With shared access key authentication, you indeed do not require a user or a claim.  
However, I once did something similar, a long time ago (2018) in asp.net core 2.
Basically , I implemented an AuthorizationHandler which took care of the verification of the required access key header.  When login was successfull, an AuthenticateResult was returned which returned  a dummy 'ClaimsPrincipal'.

Maybe you can consider this overkill for such a 'simple' feature.